opengrok-indexer/src/main/java/org/opengrok/indexer/history/BazaarHistoryParser.java

```diff
@@ -110,79 +111,81 @@ public void processStream(InputStream input) throws IOException {
                 continue;
             }
 
-            switch (state) {
-                case 0:
-                    // First, go on until revno is found.
-                    if (s.startsWith("revno:")) {
-                        String[] rev = s.substring("revno:".length()).trim().split(" ");
-                        entry.setRevision(rev[0]);
-                        ++state;
-                    }
-                    break;
-                case 1:
-                    // Then, look for committer.
-                    if (s.startsWith("committer:")) {
-                        entry.setAuthor(s.substring("committer:".length()).trim());
-                        ++state;
-                    }
-                    break;
-                case 2:
-                    // And then, look for timestamp.
-                    if (s.startsWith("timestamp:")) {
-                        try {
-                            Date date = repository.parse(s.substring("timestamp:".length()).trim());
-                            entry.setDate(date);
-                        } catch (ParseException e) {
-                            //
-                            // Overriding processStream() thus need to comply with the
-                            // set of exceptions it can throw.
-                            //
-                            throw new IOException("Failed to parse history timestamp:" + s, e);
+            if (Objects.nonNull(entry)) {
```

Comment:
what is the advantage of using `Objects.nonNull` compared to simple `!= null` check ? Also, this is adding extra level of indentation for a giant switch/case block. Can this be avoided by refactoring the block into a method or by `continue` ?

Reply:
Objects.nonNull is same as !=null, AAdvantage is while using in streams.
ex: testlList.stream().filter(Objects::nonNull) vs testlList.stream().filter(elem -> elem!=null)

Reply:
Skip this swtch case to multiple methods